File: Semester 2/Object Oriented Programming/Labs/Lab_Test2_Practice/build_your_car/car.cpp
#include "car.h"
using namespace std;

double Car::computePrice() {
  if (bodyStyle == "Sedan") {
    return 8000 + engine->getPrice();
  } else { // Convertible
    return 9000 + engine->getPrice();
  }
}

string Car::toString() {
  return "Body Style: " + bodyStyle + " | " + engine->toString() + " | Price: " + to_string(computePrice()) + "\n";
}

Engine* Car::getEngine() {
  return engine;
}

File: Semester3/Advanced_Programming_Methods/Lab/Lab2/A1/src/main/java/org/example/Model/Car.java
package org.example.Model;

public class Car implements IEntity {
    private int price;
    public Car(int price) {
        this.price = price;
    }
    public int getPrice() {
        return price;
    }

    public void setPrice(int price) {
        this.price = price;
    }

    @Override
    public int computePrice() {
        return price;
    }

    @Override
    public boolean equals(Object other) {
        if(!(other instanceof Car car)) {
            return false;
        }
        return car.getPrice() == this.price;
    }

    @Override
    public String toString() {
        return "Car: " + this.price + " RON";
    }

    @Override
    public String myToString() {
        return this.toString();
    }
}
